modules/objdetect/src/qrcode.cpp

```diff
@@ -6,23 +6,1220 @@
+    std::string f;
+    int num = total;
+    for(int i=format;num>0;i=i>>1,num--){
+        if(i%2==1)
+            f='1'+f;
+        else
+            f='0'+f;
+    }
+    return f;
+}
+const char * getSrcMode(const int& eci_mode){
+    switch (eci_mode){
+        case CP437:
+        case CP437_:
+            return "CP437";
+        case ISO_8859_1:
+        case ISO_8859_1_:
+            return "ISO-8859-1";
+        case ISO_8859_2:
+            return "ISO-8859-2";
+        case ISO_8859_3:
+            return "ISO-8859-3";
+        case ISO_8859_4:
+            return "ISO-8859-4";
+        case ISO_8859_5:
+            return "ISO-8859-5";
+        case ISO_8859_6:
+            return "ISO-8859-6";
+        case ISO_8859_7:
+            return "ISO-8859-7";
+        case ISO_8859_8:
+            return "ISO-8859-8";
+        case ISO_8859_9:
+            return "ISO-8859-9";
+        case ISO_8859_10:
+            return "ISO-8859-10";
+        case ISO_8859_11:
+            return "ISO-8859-11";
+        case ISO_8859_13:
+            return "ISO-8859-13";
+        case ISO_8859_14:
+            return "ISO-8859-14";
+        case ISO_8859_15:
+            return "ISO-8859-15";
+        case ISO_8859_16:
+            return "ISO-8859-16";
+        case Shift_JIS:
+            return "SHIFT_JIS";
+        case CP1250:
+            return "CP1250";
+        case CP1251:
+            return "CP1251";
+        case CP1252:
+            return "CP1252";
+        case CP1256:
+            return "CP1256";
+        case UTF_16BE:
+            return "UTF−16BE";
+        case UTF_8:
+            return "UTF−8";
+        case US_ASCII:
+            return "ASCII";
+        case Big5:
+            return "BIG5";
+        case GBK:
+            return "GBK";
+        case EUC_KR:
+            return "EUC−KR";
+        default:
+            return "UTF−8";
+    }
+}
+/**
+ *params@  cur_str cur_str_len input str
+ *func@ add a str into the cur_str
+ */
+void loadString(const std::string &str ,vector<uint8_t>& cur_str,bool is_bit_stream = false){
+    for(size_t i = 0; i < str.length();i++){
+        if(is_bit_stream){
+            cur_str.push_back(uint8_t (str[i]-'0'));
+        }
+        else
+            cur_str.push_back(uint8_t (str[i]));
+    }
+    return;
+}
+
+/*total codewords are divided into two groups
+ *The ecc_codewords are the same in two groups*/
+struct BlockParams {
+    int ecc_codewords;           //Number of error correction blocks
+    int num_blocks_in_G1;
+    int data_codewords_in_G1;
+    int num_blocks_in_G2;
+    int data_codewords_in_G2;
+};
+
+struct VersionInfo {
+    int	total_codewords;
+    int	alignment_pattern[max_alignment];  //(location of alignment pattern)
+    BlockParams ecc[4];
+};
+struct DataOfAI{
+    int Data_len;
+    bool fixed_len;
+};
+struct AIinGS1{
+    std::string AI_name;
+    int AI_len;
+    DataOfAI data[2];
+    std::string data_title;
+};
+
+const AIinGS1 GS1_AI_database[999] = {
```

Comment:
I haven't written all of the GS1_AI . I will complete that later.

Reply:
I think these values are not needed, because output interpretation is not the task of decoder. Can we use just string length (for FNC in first position) instead of data length given in this array?